core/vm/evm.go

```diff
@@ -536,26 +537,43 @@ func (evm *EVM) TobinTransfer(db StateDB, sender, recipient common.Address, gas
 
 	if amount.Cmp(big.NewInt(0)) != 0 {
 		reserveAddress, err := GetRegisteredAddressWithEvm(params.ReserveRegistryId, evm)
-		if err != nil && err != errors.ErrSmartContractNotDeployed && err != errors.ErrRegistryContractNotDeployed {
+		if err != nil && err != contract_errors.ErrSmartContractNotDeployed && err != contract_errors.ErrRegistryContractNotDeployed {
 			log.Trace("TobinTransfer: Error fetching Reserve address", "error", err)
 		}
 
 		if err == nil {
```

Comment:
nit: readability

Instead of 3 nested `ifs` you can do:

```go
if err != nil {
  return
}
//...continue
```

Reply:
I'm actually not sure I can, since we don't want to exit the function if there's an error. Instead, we want to complete the transfer *without* charging the tobin tax.